util/dvsim/Deploy.py

```diff
@@ -561,17 +564,35 @@ def post_finish(self, status):
             rm_path(self.cov_db_test_dir)
 
     @staticmethod
-    def get_seed():
+    def get_test_seed() -> (int, int, int):
+        '''Get the seed to use for a test, as three forms: base, short, long.
+
+        The base seed is the one that the user might want to supply, and the
+        other two seeds are derived from it. The short seed is guaranteed to be
+        usable as a SystemVerilog seed, so is not more than 2^32. The long seed
+        is generated from the base see
+        '''
+
         # If --seeds option is passed, then those custom seeds are consumed
         # first. If --fixed-seed <val> is also passed, the subsequent tests
         # (once the custom seeds are consumed) will be run with the fixed seed.
-        if not RunTest.seeds:
-            if RunTest.fixed_seed:
-                return RunTest.fixed_seed
+        base = None
+        if RunTest.seeds:
+            base = RunTest.seeds.pop(0)
+        elif RunTest.fixed_seed is not None:
+            base = RunTest.fixed_seed
+        else:
             for i in range(1000):
-                seed = random.getrandbits(256)
-                RunTest.seeds.append(seed)
-        return RunTest.seeds.pop(0)
+                base = random.getrandbits(16)
```

Comment:
I thought you intended to use 32 bits?

Reply:
It definitely mustn't be more than 32 bits, but I'm unconvinced making it that long is necessary (and the shorter options are easier to type / remember...). 65k options is still quite a lot! What do you think?